docs/strategy-advanced.md

```diff
@@ -42,32 +42,34 @@ class AwesomeStrategy(IStrategy):
 
 ***
 
-### Storing custom information using DatetimeIndex from `dataframe`
+## Custom sell signal
 
-Imagine you need to store an indicator like `ATR` or `RSI` into `custom_info`. To use this in a meaningful way, you will not only need the raw data of the indicator, but probably also need to keep the right timestamps.
+It is possible to define custom sell signals. This is very useful when we need to customize sell conditions for each individual trade.
 
-```python
-import talib.abstract as ta
-class AwesomeStrategy(IStrategy):
-    # Create custom dictionary
-    custom_info = {}
+An example of how we can set stop-loss and take-profit targets in the dataframe and also sell trades that were open longer than 1 day:
 
-    def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
-        # using "ATR" here as example
-        dataframe['atr'] = ta.ATR(dataframe)
-        if self.dp.runmode.value in ('backtest', 'hyperopt'):
-          # add indicator mapped to correct DatetimeIndex to custom_info
-          self.custom_info[metadata['pair']] = dataframe[['date', 'atr']].set_index('date')
-        return dataframe
+``` python
+class AwesomeStrategy(IStrategy):
+    def custom_sell(self, pair: str, trade: Trade, current_time: datetime, current_rate: float,
+                    current_profit: float, dataframe: Dataframe, **kwargs) -> bool:
+        trade_row = dataframe.loc[dataframe['date'] == timeframe_to_prev_date(trade.open_date_utc)].squeeze()
+
+        # Sell when price falls below value in stoploss column of taken buy signal.
+        if 'stop_loss' in trade_row:
+            if current_rate <= trade_row['stop_loss'] < trade.open_rate:
+                return 'stop_loss'
```

Comment:
Yeah its just what i lifted from my experiments strategy. Might not be the best thing to include in docs. However this kind of usage made me think about it from a bit different angle.

I find it _extremely_ convenient to have absolute stoploss and take-profit (or roi) values in dataframe, having them calculated for each buy signal. This would be extremely handy for less-technical users. This does not work well with backtesting however. Might be a feature worthy of another PR.

Reply:
i'm not sure why / what would not work well for backtesting?

Reply:
Backtester has no notion of stoploss or roi from these functions. What we do is basically emit a sell signal, which follows sell signal rules.

> Sell-signal sells happen at open-price of the consecutive candle

So suppose we have a current in-progress candle that reaches take-profit target, but then falls down and closes below. Since a wick touched our take-profit target a dry run would try to sell that price (correct me if im wrong here). But in backtest sell signal would not even happen because candle closed below take-profit target. Same is true for stoploss. Actually `custom_sell` for stoploss seems like a very bad idea. I think we should purge it from docs. It is both inaccurate and too slow (no stoploss order placed on exchange).

Reply:
well yes, sell-signal and stop / trailing stoploss do serve a different purpose for sure, while the custom_stop function allows you to trail the current price (avoid large pullbacks / sells on large pullbacks) - also with stoploss on exchange in place - custom-sell will definitely not allow this (it's also not the purpose of this function).

you can however still have multiple different sell-reasons - like - you "notice" that the pattern you entered on does not play out - so you bail out even at a loss (avoiding a greater loss via Stoploss).

> So suppose we have a current in-progress candle that reaches take-profit target, but then falls down and closes below.

This is an assumption about movements backtesting does not know about - maybe it reached the take-profit target first, then fell - but maybe it fell first, and then reached the take-profit target (you'd be out of the trade in dry-run in that case too).
Obviously it's an assumption backtesting makes (and is documented), and it's a totally different argument if we assume "price went to high first" or "price went to low first" (green / red candles will not determine this ...)

Reply:
I am not having any assumptions where price went first though. Consider the situation where wick touches take-profit target, but candle does not hit stoploss. It does not matter which way price went first because if it went down - nothing would happen anyway, but if it went up - we can take profit. There is no issue with backtesting assumptions at all, its just that such columns could act more like roi and stoploss instead of sell signal.

Reply:
i'm ... not quite certain what you're referring to with this (probably we should discuss this in chat, as i don't THINK it's related to this PR directly).

If ROI happens (minimal ROI) - then the ROI value is assumed as sell price - not high or low or open.
it's similar with stoploss ... so this column (virtually) exists already ...

Reply:
I do agree with you though - i think this example should be ... more generic - at least it should not use stoploss / take_profit directly, as that might be missleading.

Reply:
I've updated the sample now to not use take-profit / stoploss, but to simply use different indicators.

I'm sure others will come up with additional / better samples ... but i think this will do for now.